### ISSUE TITLE
TextInput module should not enforce type attribute

### ISSUE THREAD

**Issue by treyturner** [geb/issues#643]
[Mozilla docs](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#input_types) state that `input`s without a `type` attribute are assumed to be `text`. It seems the `TextInput` module then should not actively enforce this by throwing exceptions like these, which I'm receiving from 5.0-milestone-2:

```
geb.error.InvalidModuleBaseException: Specified base element for geb.module.TextInput module was an input of type '' but only input of type text is allowed as the base element.
```

This issue raised by request after a brief discussion on the Geb Slack. Thank you for your time and the lovely framework!

**Comment by erdi** [comment]
So I've looked into this and this will most likely be browser specific. You can see in [`TextInputSpec`](https://github.com/geb/geb/blob/6f621cd30d28310ced63dae33b4e7e012d900cea/module/geb-core/src/test/groovy/geb/module/TextInputSpec.groovy#L22) that the html setup in the test does not specify a `type` attribute yet the tests are passing. I checked and at least on HtmlUnit, a version of Chrome and a version of Firefox getting the `type` attribute on an `input` that does not specify that attribute returns `text`. Can you therefore please tell me which operating system, browser and browser version did you observe this failure on, @treyturner ?

**Comment by treyturner** [comment]
I reproduced this again just now on Windows 10 1909 (18363.1646) with Chrome 92.0.4515.131 and Chromedriver 91.0.4472.19. When I first experienced it, I'm sure I was on Chrome 91 though. The problem seems to relate to this specific version of Chromedriver, because when I updated just now to 91.0.4472.101 the problem goes away. I wonder if it somehow relates to [this Chromedriver bug](https://bugs.chromium.org/p/chromium/issues/detail?id=1205107)?

In any case, thank you for the diligence! I'm going to close this out because I don't think there's anything for Geb to fix or improve - in the legitimate cases this exception would be thrown, a different input type has probably been selected and the message would be appropriate and helpful.
